Userland/DevTools/UserspaceEmulator/Emulator.cpp

```diff
@@ -215,21 +217,85 @@ int Emulator::exec()
 
     while (!m_shutdown) {
         m_cpu.save_base_eip();
-
         auto insn = X86::Instruction::from_stream(m_cpu, true, true);
+        if (m_steps_til_pause) {
+            // Exec cycle
+            if constexpr (trace) {
+                outln("{:p}  \033[33;1m{}\033[0m", m_cpu.base_eip(), insn.to_string(m_cpu.base_eip(), symbol_provider));
+            }
 
-        if constexpr (trace) {
-            outln("{:p}  \033[33;1m{}\033[0m", m_cpu.base_eip(), insn.to_string(m_cpu.base_eip(), symbol_provider));
-        }
+            (m_cpu.*insn.handler())(insn);
+
+            if constexpr (trace) {
+                m_cpu.dump();
+            }
 
-        (m_cpu.*insn.handler())(insn);
+            if (m_pending_signals) [[unlikely]] {
+                dispatch_one_pending_signal();
+            }
+            if (m_steps_til_pause > 0)
+                m_steps_til_pause--;
+
+        } else {
+            // Console interface
+            // FIXME: Previous Instruction**s**
+            // FIXME: Function names (base, call, jump)
+            auto eip = m_cpu.base_eip();
+            outln("==> {:p}: {}", m_cpu.base_eip(), insn.to_string(m_cpu.base_eip()));
+            for (int i = 0; i < 7; ++i) {
+                m_cpu.save_base_eip();
+                insn = X86::Instruction::from_stream(m_cpu, true, true);
+                outln("    {:p}: {}", m_cpu.base_eip(), insn.to_string(m_cpu.base_eip()));
+            }
+            // We don't want to increase EIP here, we just want the instructions
+            m_cpu.set_eip(eip);
 
-        if constexpr (trace) {
+            outln();
             m_cpu.dump();
-        }
+            outln();
+            // FIXME: this is to wordy for my taste, there should be a better
+            //        way of doing this; The whole REPL is jank anyway...
+            //        LibLine could be a solution for this
+            auto line_or_error = m_editor->get_line(">> ");
+            if (line_or_error.is_error())
+                continue;
+
+            auto help = [] {
+                outln("Available commands:");
+                outln("continue, c: Continue the exectution");
```

Comment:
exectution -> execution

Reply:
I don't know why the spell checker dindn't catch that the only thing I could find is ex[s]ection, meaning amputation....

fixed either wasy